esx_service/utils/auth_data.py

```diff
@@ -561,11 +568,43 @@ def handle_upgrade_1_1_to_1_2(self):
                      """
             sql_script = script.format(DB_MAJOR_VER, DB_MINOR_VER)
             self.conn.executescript(sql_script)
+
+            logging.debug("handle_upgrade_1_1_to_1_2: update vms table Done")
+
+            # update the tenants table to set "default_datastore" to "__VM_DS" if "default_datastore" is ""
+            self.conn.execute("UPDATE OR IGNORE tenants SET default_datastore_url = ?  where default_datastore_url = \"\"",
+                              (auth_data_const.VM_DS_URL,))
+            logging.debug("handle_upgrade_1_1_to_1_2: update default_datastore in tenants table")
+
+            cur = self.conn.execute("SELECT * FROM tenants")
+            result = cur.fetchall()
+
+            self.conn.execute("INSERT OR IGNORE INTO privileges(tenant_id, datastore_url, allow_create, max_volume_size, usage_quota)"
+                                 "SELECT tenants.id, tenants.default_datastore_url, 1, 0, 0 FROM tenants")
```

Comment:
Nit: just for readability, a single string is nicer. 
"""INSERT  or IGNORE ....
      SELECT ..."""